README.md

```diff
@@ -1,42 +1,211 @@
+
+Generate an installer ISO image.  This will contain the following:
+
+* `preseed.cfg` - From `debian-<release>/preseed.cfg`.
+* `sdc-vmtools` - The current content of the
+  [sdc-vmtools](https://github.com/joyent/sdc-vmtools) repo.
+* `early-command` - A preinstallation script, from
+  `debian-<release>/early-command`
+* `late-command` - A postinstallation script, from
+  `debian-<release>/late-command`
+
+This image is automatically mounted when the debian installer needs it, but is
+not automtically mounted in the chroot (`/target`) while `late-command` is
+running.  For that reason, each release's late command needs to mount it at
+`/target/cdrom` prior to running `late-command`.
+
+
+### image
+
+This runs qemu in a way that allows unattended installation using the media
+described above.  Once qemu exits, a Triton-compatible
+image is generated and stored in the current `bits/<stamp>` directory as
+`debian-<release>-<timestamp>.{json,tar.gz}`.
+
+The actual image creation is handled by `sdc-vmutils/bin/create-hybrid-iamge`.
 
-## Using
 
-Once configured, run the following command:
+### upload
 
+This invokes
+[bits-upload.sh](https://github.com/joyent/eng/blob/master/tools/bits-upload.sh)
+to upload the image to manta and updates.joyent.com.
+
+
+## Default Settings For Images
+
+Each image has the following characteristics.  See
+`debian-<release>/ks.cfg` for details on which packages are included.
+
+* Disk is 10GB in size (8GB for / and the rest for swap)
+* Stock Kernel
+* US Keyboard and Language
+* Firewall enabled with SSH allowed
+* Timezone is set to UTC
+* Console is on ttyS0
+* Root password is blank: console login is allowed without a password
+* Configuration from the SmartOS metadata service is performed using cloud-init.
+
+## Development
+
+The following serves as a guide for adding support for new RHEL-like
+distributions and versions of existing distributions.
+
+Distribution-specific content is found in a per-distro subdirectory.  For
+example, Debian 10 bits are in the `debian-10` directory.  Directory names are
+always lower-case.
+
+The following subsections describe the content that may be in a per-distro
+directory.
+
+### preseed.cfg file
+
+The [preseed](https://wiki.debian.org/DebianInstaller/Preseed) configuration
+file.  Notable parts of this include:
+
+* `cloud-init` is installed, as it is responsible for interacting with the
+  host's metadata service to configure networking, run user scripts, etc.  It
+  requires `pyserial`, but for "reasons" the cloud-init developers have avoided
+  adding pyserial as a dependency.
+* `cloud-init` requires configuration in
+  `/etc/cloud/cloud.cfg.d/90\_smartos.cfg` to only enable the SmartOS
+  datasource, among other things.
+* A `preseed/early\_command` (`debian-<release>/early-command`) is used to tail
+  the installation log and write it to `/dev/ttyS0`.  This script is run under
+  busybox, so avoid use of anything too fancy.  `qemu` runs in such a way that
```

Comment:
Nit 'so avoid use' sounds a bit awkward (maybe "so avoid the use" ?)

Reply:
ok